docs/modules/cosmosdb.md

```diff
@@ -27,6 +27,10 @@ Provisions a new CosmosDB database account that can be accessed through any of t
 | `ipFilters.allowAccessFromAzure` | `string` | Specifies if Azure Services should be able to access the CosmosDB account.Valid valued are `""` (unspecified), `enabled`, or `disabled`. | N | If left unspecified, defaults to enabled. |
 | `ipFilters.allowAccessFromPortal` | `string` | Specifies if the Azure Portal should be able to access the CosmosDB account. If `allowAccessFromAzure` is set to enabled, this value is ignored. Valid valued are `""` (unspecified), `enabled`, or `disabled`. | N | If left unspecified, defaults to enabled. |
 | `ipFilters.allowedIPRanges` | `array` | Values to include in IP Filter. Can be IP Address or CIDR range. | N | If not specified, no additional values will be included in filters. |
+| `consistencyPolicy` | `object` | The consistency policy for the Cosmos DB account. | N | |
+| `consistencyPolicy.defaultConsistencyLevel` | `string` | The default consistency level and configuration settings of the Cosmos DB account. - Eventual, Session, BoundedStaleness, Strong, ConsistentPrefix | Y | |
+| `consistencyPolicy.maxStalenessPrefix` | `integer` | When used with the Bounded Staleness consistency level, this value represents the number of stale requests tolerated. Accepted range for this value is 1 – 2,147,483,647. Required when defaultConsistencyPolicy is set to 'BoundedStaleness'. | N | |
```

Comment:
Putting an idea out there...

For many things I have worked on, if an option is subordinate in some way to some other option, I try to make that scope clear through the keys...

For example, in many k8s charts, you can override `service.type`. Often if type `NodePort` is selected, other options become available (like choosing a specific port), but they'd applicable _only_ when `service.type: NodePort`. e.x.:

```
service:
  ## Type of service; valid values are "ClusterIP", "LoadBalancer", and
  ## "NodePort". "ClusterIP" is sufficient in the average case where a service
  ## catalog installation in the same cluster is the only client that needs to
  ## communicate with this broker. i.e. The broker does not need to receive
  ## inbound requests from outside the cluster.
  type: ClusterIP
  ## Further configuration if service is of type "NodePort"
  nodePort:
    ## Available port in allowable range (e.g. 30000 - 32767 on minikube)
    port: 30080
```

I'm wondering if the same sort of strategy would add some clarity here. What if `consistencyPolicy. maxStalenessPrefix ` were changed to `consistencyPolicy.boundedStateless.maxStalenessPrefix`. Would that make it more clear that the option is applicable only when `consistencyPolicy.defaultConsistencyLevel: BoundedStaleness`?

I'm not strongly encouraging this. I'm only opening this up to discussion. Do not count this as a blocker by any means.

Reply:
Yeah I think so. See my other comment, I mis-read the docs before!